src/component/contentpost/index.tsx

```diff
@@ -141,9 +148,11 @@ function ContentPost(props: ContentPostProps) {
         count: count,
         text2: count === 1 ? "reply" : "replies",
         onPress: () => {
-          if (isVisible === "postDetails") {
+          if (isVisible === "postDetails" && post?.id === id) {
```

Comment:
Here's the main change. If the post details are already open, close them. Otherwise, open them.